### DISCUSSION TITLE
cannot be installed

### DISCUSSION THREAD

**Discussion by 57OF4R0** [PabloLec/RecoverPy#168] [General]
I'm sorry for such stupid questions, but I'm new to this, and I can't install your application, at the installation stage, writing the command (python3 -m pip install recoverpy) I get errors in the console.
Thank you in advance.
<pre>root@amnesia:~# python3 -m pip install recoverpy
<font color="#A2734C">WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by &apos;NewConnectionError(&apos;&lt;urllib3.connection.HTTPSConnection object at 0x7eeeba57c760&gt;: Failed to establish a new connection: [Errno -3] Временный сбой в разрешении имен&apos;)&apos;: /simple/recoverpy/</font>
<font color="#A2734C">WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by &apos;NewConnectionError(&apos;&lt;urllib3.connection.HTTPSConnection object at 0x7eeeba57c2b0&gt;: Failed to establish a new connection: [Errno -3] Временный сбой в разрешении имен&apos;)&apos;: /simple/recoverpy/</font>
<font color="#A2734C">WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by &apos;NewConnectionError(&apos;&lt;urllib3.connection.HTTPSConnection object at 0x7eeeba57c100&gt;: Failed to establish a new connection: [Errno -3] Временный сбой в разрешении имен&apos;)&apos;: /simple/recoverpy/</font>
<font color="#A2734C">WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by &apos;NewConnectionError(&apos;&lt;urllib3.connection.HTTPSConnection object at 0x7eeeba57c4f0&gt;: Failed to establish a new connection: [Errno -3] Временный сбой в разрешении имен&apos;)&apos;: /simple/recoverpy/</font>
<font color="#A2734C">WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by &apos;NewConnectionError(&apos;&lt;urllib3.connection.HTTPSConnection object at 0x7eeeba6c9550&gt;: Failed to establish a new connection: [Errno -3] Временный сбой в разрешении имен&apos;)&apos;: /simple/recoverpy/</font>
<font color="#C01C28">ERROR: Could not find a version that satisfies the requirement recoverpy</font>
<font color="#C01C28">ERROR: No matching distribution found for recoverpy</font>
</pre>

**Comment by PabloLec** [comment]
Hello,

It might be that you're using an old python version. Although RecoverPy v1 supported Python 3.6 and I guess pip should default to this version if you had it. FYI RecoverPy v2 only support Python >= 3.8. Python 3.6 reached its EOL a year ago and 3.7 will reach its EOL in a few months.

You can try running `python3 --version`, and you should probably update your Python version if needed.

If your Python version is not the source of the problem, it might be that your Python and/or pip bin defaults to the wrong installation. Try running : 

`pip3 install recoverpy`
or
`python3 -m pip3 install recoverpy`

**Reply by 57OF4R0** [reply]
Version Python 3.9.2

via the command: (pip3 install recoverpy) gives the same errors
and the second team produces this: `root@amnesia:~# python3 -m pip3 install recoverpy
/usr/bin/python3: No module named pip3`

**Reply by 57OF4R0** [reply]
Version Pip3:
```
root@amnesia:~# pip3 --version
pip 20.3.4 from /usr/lib/python3/dist-packages/pip (python 3.9)
```

**Reply by PabloLec** [reply]
Oh I didn't pay attention but you have connection errors when running pip. So it's probably a network error.

You should try to dig into it, links:
https://stackoverflow.com/questions/52815784/python-pip-raising-newconnectionerror-while-installing-libraries
https://stackoverflow.com/questions/67138344/python-pip-raising-newconnectionerror-while-installing-django-braces
https://stackoverflow.com/questions/59635259/pip-install-generates-warnings-newconnectionerrorpip-vendor-urllib3-conne

**Reply by 57OF4R0** [reply]
Please excuse me, but I can't figure out how to get the rights to: `resolv.conf` so that I can add public DNS

**Reply by 57OF4R0** [reply]
In general, I figured out my update problem, but now when I try to search, when I enter what to search for on select yes search and click search, a window with copies (open, exit) appears for a moment and gives another error:
`CalledProcessError: Command '['blockdev', '--getbsz', '/dev/sda3']' returned non-zero exit status 1.
`
I would be very grateful if you could tell me what to do.

**Reply by PabloLec** [reply]
Are your running RecoverPy as root or using sudo ?

**Reply by 57OF4R0** [reply]
I have a question, so that maybe I didn't even try to do something already, I had photos in a hidden folder, and then I deleted it, tried to restore through Windows applications, but not one photo was ever recovered, so maybe I shouldn't try to restore through your program on linux?

**Reply by PabloLec** [reply]
Yeah RecoverPy is more string based research oriented. You could still try to recover a photo or a binary file but it would be very difficult if it doesn't contain any possible string to search.
You should probably try PhotoRec on Windows which is designed to do that.

**Reply by 57OF4R0** [reply]
Unfortunately, I've already used PhotoRec, but it didn't give me the results I needed

**Reply by 57OF4R0** [reply]
My friend, I managed to install everything and run it on Linux Mint, hopefully the search will give me results, thanks for answering me.
Linux is an addictive thing, I've never used it before!
Good luck to you!
